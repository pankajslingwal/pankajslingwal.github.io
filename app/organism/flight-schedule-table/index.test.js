import React from 'react';
import renderer from 'react-test-renderer';
import FlightScheduleTable from './';
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";


describe("Flight Schedule Table", () => {

    it('renders correctly', () => {
        const tree = renderer.create(<FlightScheduleTable />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('After row selected, Select Flight button disables and row is highlighted', () => {
        render(<FlightScheduleTable />);
        userEvent.click(screen.getAllByText("Select Flight")[0]);

        const btn = screen.getAllByText("Select Flight")[0];
        const row = screen.getAllByRole("button")[0];
        
        expect(btn.className).toBe("disabled");
        expect(row.className).toContain("schedule-row selected-row");
    });

    it('After another row selected, Select Flight button disables and row is highlighted and revert previous row', () => {
        render(<FlightScheduleTable />);
        
        //Click of 1st row button
        userEvent.click(screen.getAllByText("Select Flight")[0]);

        const btn1 = screen.getAllByText("Select Flight")[0];
        const row1 = screen.getByTestId("schedule-row-0");
        
        //1st row button to be disabled and row highlighted
        expect(btn1.className).toBe("disabled");
        expect(row1.className).toContain("schedule-row selected-row");

        const btn2 = screen.getAllByText("Select Flight")[1];
        const row2 = screen.getByTestId("schedule-row-1");
        
        //2nd row button to be enabled and row not highlighted
        expect(btn2.className).toBe("");
        expect(row2.className.trim()).toBe("schedule-row");

        //After click of 2nd row button
        userEvent.click(screen.getAllByText("Select Flight")[1]);

        //1st row button to be enabled and row not highlighted
        expect(btn1.className).toBe("");
        expect(row1.className).toContain("schedule-row");

        //2nd row button to be disabled and row highlighted
        expect(btn2.className).toBe("disabled");
        expect(row2.className.trim()).toBe("schedule-row selected-row");
    });

});