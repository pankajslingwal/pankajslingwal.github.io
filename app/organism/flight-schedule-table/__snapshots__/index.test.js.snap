// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight Schedule Table renders correctly 1`] = `
<div
  className="content"
>
  <ul
    className="schedule-header"
  >
    <li>
      Depart
    </li>
    <li>
      Arrive
    </li>
    <li>
      Flight
    </li>
    <li>
      Total Duration
    </li>
    <li>
      Â 
    </li>
  </ul>
  <ul
    aria-label="Flight Air India (AI0234) departs from DXB at 09:30 and arrives BOM at 13:20, total Flight duration is 3 hours 30 minutes."
    className="schedule-row "
    data-testid="schedule-row-0"
    role="button"
    tabIndex="0"
  >
    <li
      className="only-desktop"
    >
      <span
        className="highlight"
      >
        09:30
      </span>
       (
      DXB
      )
    </li>
    <li
      className="only-desktop"
    >
      <span
        className="highlight"
      >
        13:20
      </span>
       (
      BOM
      )
    </li>
    <li
      className="only-mobile"
    >
      <span
        className="highlight"
      >
        09:30
         - 
        13:20
      </span>
    </li>
    <li
      className="only-mobile"
    >
      <span>
        DXB
         - 
        BOM
      </span>
    </li>
    <li
      className="only-mobile"
    >
      3 hours 30 minutes
    </li>
    <li>
      <span
        className="color-highlight icon "
      >
        <img
          alt="Flight Air India (AI0234) logo"
          src="test-file-stub"
        />
        Air India (AI0234)
      </span>
    </li>
    <li
      className="only-desktop"
    >
      3 hours 30 minutes
    </li>
    <li>
      <button
        aria-label="Select Flight number Air India (AI0234)"
        aria-pressed={false}
        className=""
        onClick={[Function]}
        title="Select Flight number Air India (AI0234)"
        type="submit"
      >
        Select Flight
      </button>
    </li>
  </ul>
  <ul
    aria-label="Flight Indigo (6E22) departs from DXB at 01:30 and arrives DEL at 03:50, total Flight duration is 2 hours 30 minutes."
    className="schedule-row "
    data-testid="schedule-row-1"
    role="button"
    tabIndex="0"
  >
    <li
      className="only-desktop"
    >
      <span
        className="highlight"
      >
        01:30
      </span>
       (
      DXB
      )
    </li>
    <li
      className="only-desktop"
    >
      <span
        className="highlight"
      >
        03:50
      </span>
       (
      DEL
      )
    </li>
    <li
      className="only-mobile"
    >
      <span
        className="highlight"
      >
        01:30
         - 
        03:50
      </span>
    </li>
    <li
      className="only-mobile"
    >
      <span>
        DXB
         - 
        DEL
      </span>
    </li>
    <li
      className="only-mobile"
    >
      2 hours 30 minutes
    </li>
    <li>
      <span
        className="color-highlight icon "
      >
        <img
          alt="Flight Indigo (6E22) logo"
          src="test-file-stub"
        />
        Indigo (6E22)
      </span>
    </li>
    <li
      className="only-desktop"
    >
      2 hours 30 minutes
    </li>
    <li>
      <button
        aria-label="Select Flight number Indigo (6E22)"
        aria-pressed={false}
        className=""
        onClick={[Function]}
        title="Select Flight number Indigo (6E22)"
        type="submit"
      >
        Select Flight
      </button>
    </li>
  </ul>
  <ul
    aria-label="Flight Emirates (EK512) departs from DXB at 12:10 and arrives BOM at 13:15, total Flight duration is 5 hours 30 minutes."
    className="schedule-row "
    data-testid="schedule-row-2"
    role="button"
    tabIndex="0"
  >
    <li
      className="only-desktop"
    >
      <span
        className="highlight"
      >
        12:10
      </span>
       (
      DXB
      )
    </li>
    <li
      className="only-desktop"
    >
      <span
        className="highlight"
      >
        13:15
      </span>
       (
      BOM
      )
    </li>
    <li
      className="only-mobile"
    >
      <span
        className="highlight"
      >
        12:10
         - 
        13:15
      </span>
    </li>
    <li
      className="only-mobile"
    >
      <span>
        DXB
         - 
        BOM
      </span>
    </li>
    <li
      className="only-mobile"
    >
      5 hours 30 minutes
    </li>
    <li>
      <span
        className="color-highlight icon "
      >
        <img
          alt="Flight Emirates (EK512) logo"
          src="test-file-stub"
        />
        Emirates (EK512)
      </span>
    </li>
    <li
      className="only-desktop"
    >
      5 hours 30 minutes
    </li>
    <li>
      <button
        aria-label="Select Flight number Emirates (EK512)"
        aria-pressed={false}
        className=""
        onClick={[Function]}
        title="Select Flight number Emirates (EK512)"
        type="submit"
      >
        Select Flight
      </button>
    </li>
  </ul>
</div>
`;
